fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/insert/GroupCommitInsertExecutor.java

```diff
@@ -95,26 +107,39 @@ private static boolean isGroupCommitAvailablePlan(PhysicalOlapTableSink<? extend
-            PhysicalOlapTableSink<?> physicalOlapTableSink)
-            throws UserException, RpcException, TException, ExecutionException, InterruptedException {
+    private static void handleGroupCommit(ConnectContext ctx, DataSink sink,
+            PhysicalOlapTableSink<?> physicalOlapTableSink, NereidsPlanner planner)
+                throws UserException, TException, RpcException, ExecutionException, InterruptedException {
         // TODO we should refactor this to remove rely on UnionNode
         List<InternalService.PDataRow> rows = new ArrayList<>();
-        List<List<Expr>> materializedConstExprLists = ((UnionNode) sink.getFragment()
-                .getPlanRoot()).getMaterializedConstExprLists();
-        int filterSize = 0;
-        for (Slot slot : physicalOlapTableSink.getOutput()) {
-            if (slot.getName().contains(Column.DELETE_SIGN)
-                    || slot.getName().contains(Column.VERSION_COL)) {
-                filterSize += 1;
-            }
+
+        Optional<PhysicalUnion> union = planner.getPhysicalPlan()
+                .<Set<PhysicalUnion>>collect(PhysicalUnion.class::isInstance).stream().findAny();
+        List<List<NamedExpression>> constantExprsList = null;
+        if (union.isPresent()) {
+            constantExprsList = union.get().getConstantExprsList();
+        }
+        Optional<PhysicalOneRowRelation> oneRowRelation = planner.getPhysicalPlan()
+                .<Set<PhysicalOneRowRelation>>collect(PhysicalOneRowRelation.class::isInstance).stream().findAny();
+        if (oneRowRelation.isPresent()) {
+            constantExprsList = ImmutableList.of(oneRowRelation.get().getProjects());
         }
-        for (List<Expr> list : materializedConstExprLists) {
-            rows.add(GroupCommitPlanner.getRowStringValue(list, filterSize));
+        List<String> columnNames = physicalOlapTableSink.getTargetTable().getFullSchema().stream()
+                .map(Column::getName)
+                .map(n -> n.replace("`", "``"))
+                .map(n -> "`" + n + "`")
+                .collect(Collectors.toList());
+        try {
+            for (List<NamedExpression> row : constantExprsList) {
+                rows.add(InsertUtils.getRowStringValue(row));
+            }
+        } catch (AnalysisException e) {
+            LOG.error("warning when group commit insert. {}", e);
         }
+
         GroupCommitPlanner groupCommitPlanner = EnvFactory.getInstance().createGroupCommitPlanner(
                 physicalOlapTableSink.getDatabase(),
-                physicalOlapTableSink.getTargetTable(), null, ctx.queryId(),
+                physicalOlapTableSink.getTargetTable(), columnNames, ctx.queryId(),
                 ConnectContext.get().getSessionVariable().getGroupCommit());
         PGroupCommitInsertResponse response = groupCommitPlanner.executeGroupCommitInsert(ctx, rows);
         TStatusCode code = TStatusCode.findByValue(response.getStatus().getStatusCode());
```

Comment:
in legacy, there is a retry to handle schema change when code is `DATA_QUALITY_ERROR`